addons/web/static/src/core/utils/colors.js

```diff
@@ -198,6 +198,80 @@ export function convertCSSColorToRgba(cssColor) {
     // Note: however, if ever implemented be careful of 'white'/'black' which
     // actually are color names for our color system...
 
+    // Check if cssColor is a color() functional notation allowing colorspace with implicit sRGB.
+
+    // First, define the formal syntax with regular expressions
+
+    const number = `([+-]?([0-9]+([.][0-9]*)?|[.][0-9]+))`;
+    const percentage = `(${number}%)`;
+
+    // <alpha-value> =
+    //   <number> | <percentage>
+
+    const alphaValue = `(${percentage}|${number})`;
+
+    // <predefined-rgb> =
+    //   srgb          |
+    //   srgb-linear   |
+    //   display-p3    |
+    //   a98-rgb       |
+    //   prophoto-rgb  |
+    //   rec2020
+
+    const predefinedRgb = `(?<predefined_rgb>srgb|srgb\\-linear|display\\-p3|a98\\-rgb|prophoto\\-rgb|rec2020)`;
+
+    // <predefined-rgb-params> =
+    //   <predefined-rgb> [ <number> | <percentage> | none ]{3}
+
+    const predefinedRgbParams = `(${predefinedRgb})\\s+(?<c1>${alphaValue}|none)\\s+(?<c2>${alphaValue}|none)\\s+(?<c3>${alphaValue}|none)`;
+
+    // <xyz-space> =
+    //   xyz      |
+    //   xyz-d50  |
+    //   xyz-d65
+
+    const xyzSpace = `(?<xyz_space>xyz|xyz\\-d50|xyz\\-d65)`;
+
+    // <xyz-params> =
+    //   <xyz-space> [ <number> | <percentage> | none ]{3}
+
+    const xyzParams = `(${xyzSpace})\\s+(?<cc1>${alphaValue}|none)\\s+(?<cc2>${alphaValue}|none)\\s+(?<cc3>${alphaValue}|none)`;
+
+    // <colorspace-params> =
+    //   <predefined-rgb-params>  |
+    //   <xyz-params></xyz-params>
+
+    const colorspaceParams = `(?<colorspace_params>${predefinedRgbParams}|${xyzParams})`;
+
+    // <slash> = /
+
+    const slash = `/`;
+
+    // <color()> =
+    //  color( <colorspace-params> [ / [ <alpha-value> | none ] ]? )
+
+    const color = `color\\(${colorspaceParams}(\\s+${slash}\\s+(?<A>${alphaValue}|none))?\\);?`;
+    const regex = new RegExp(color);
+
+    if (regex.exec(cssColor) !== null) {
```

Comment:
The method Regex.exec() return null if it fails and 0-indexed property if it succeeds. I have to compare it to `null`

Reply:
![image](https://github.com/odoo/odoo/assets/78849981/ab4b7587-cb8b-4221-a613-f333a7504755)
So if your exec succeed, `if (regex.exec(cssColor)` will always be true. Or am I missing something ? :thinking:

Reply:
If the result is 0 (falsy) the exec() method has succeeded. So it will be a false negative. In most cases the result will be 0 (pattern matched at the start of the string), thus falsy.

Reply:
You mean the length of the array is 0 right ?
In JS `[]` is truthy.

It's ok if you leave `!== null` but it deserves a comment to say that the exec return null if it doesn't match the regex :+1:

Reply:
Right, sorry for the misunderstanding. But here the example form the same website
`const myRe = /ab*/g;
const str = "abbcdefabh";
let myArray;
while ((myArray = myRe.exec(str)) !== null) {
  let msg = `Found ${myArray[0]}. `;
  msg += `Next match starts at ${myRe.lastIndex}`;
  console.log(msg);
}`

Reply:
Or I could just use Regex.test() that returns true or false

Reply:
Done. It's better like that.